### PR TITLE
Fix TempFilter module

### PR DESCRIPTION
**Pull request by varunpuranik**
The TempFilter has a wait in the async main method. This causs the threads to block and the callback to receive messages never comes through. 
Changing that to use await.

### CODE REVIEW

**Comment by myagley** [comment]
Can you add more description about what this is fixing?

**Comment by varunpuranik** [comment]
> Can you add more description about what this is fixing?

Yes, sorry.. added now.

File: edge-modules/TemperatureFilter/Program.cs [hunk 1]
```diff
@@ -65,13 +65,13 @@ static async Task<int> MainAsync()
 
             Tuple<ModuleClient, ModuleConfig> userContext = moduleclientAndConfig;
 
-            await moduleclientAndConfig.Item1.SetInputMessageHandlerAsync("input1", PrintAndFilterMessages, userContext).ConfigureAwait(false);
+            await moduleclientAndConfig.Item1.SetInputMessageHandlerAsync("input1", PrintAndFilterMessages, userContext);
```
Comment (philipktlin) [inline edge-modules/TemperatureFilter/Program.cs]: can you help me understand the reason to remove "ConfigureAwait(false)" here?

Comment (varunpuranik) [reply edge-modules/TemperatureFilter/Program.cs]: Primarily for consistency.. we don't use it anywhere in our code.

**Pull request merged**